tests/EvmAcceptanceTests/test/ContractDeployment.js

```diff
@@ -1,50 +1,256 @@
-        const contract = await helper.deployContract("MultiParamConstructor", {
-            constructorArgs: ["saeed", 123]
+            it("Should be deployed successfully", async function () {
+                expect(contract.address).exist;
+            })
+
+            it("Should return 123 when number view function is called", async function () {
+                expect(await contract.methods.number()).to.be.eq(123)
+            })
         })
 
-        const name = await helper.getStateAsString(contract._address, 0)
-        expect(name).to.be.equal("saeed")
+        describe("When web3.js is used", function () {
+            // TODO
+        });
+    })
 
-        const number = await helper.getStateAsNumber(contract._address, 1)
-        expect(number).to.be.equal(123)
- 
-        const number2 = await contract.methods.number().call()
-        expect(number2).to.be.equal('123')
-    });
+    describe("Contract with one parameter constructor", function () {
+        describe("When ethers.js is used", function () {
+            describe("When constructor parameter is a uint256", async function () {
+                let contract;
+                let INITIAL_NUMBER = 100;
+                before(async function () {
+                    const Contract = await ethers.getContractFactory("WithUintConstructor")
+                    contract = await Contract.deploy(INITIAL_NUMBER)
+                })
 
-    it("Testing deployment of a contract with an initial fund", async function() {
-        const INITIAL_FUND = 300;
+                it("Should be deployed successfully", async function () {
+                    expect(contract.address).exist;
+                })
 
-        const contract = await helper.deployContract("ParentContract", {
-            value: INITIAL_FUND
-        })
+                it("Should return 100 when number view function is called", async function () {
+                    expect(await contract.number()).to.be.eq(INITIAL_NUMBER)
+                })
+            })
+
+            describe("When constructor parameter is a string", async function () {
+                let contract;
+                let INITIAL_NAME = "Zilliqa";
+                before(async function () {
+                    const Contract = await ethers.getContractFactory("WithStringConstructor")
+                    contract = await Contract.deploy(INITIAL_NAME)
+                })
+
+                it("Should be deployed successfully", async function () {
+                    expect(contract.address).exist;
+                })
+
+                it("Should return Zilliqa when name view function is called", async function () {
+                    expect(await contract.name()).to.be.eq(INITIAL_NAME)
+                })
+            })
+
+            // TODO
+            describe("When constructor parameter is an address", async function () {
+                it("Should be deployed successfully")
```

Comment:
there is a few empty code bits - usually good practice not to include TODOs etc

Reply:
these are pending tests supposed to be added by the rest of the team very soon :)